bindings/fcl.go

```diff
@@ -0,0 +1,77 @@
+package bindings
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/onflow/flixkit-go"
+	bindings "github.com/onflow/flixkit-go/bindings/templates"
+	v1_1 "github.com/onflow/flixkit-go/flixkitv1_1"
+)
+
+func NewFclTSGenerator() *FclGenerator {
+	templates := []string{
+		bindings.GetTsFclMainTemplate(),
+		bindings.GetTsFclScriptTemplate(),
+		bindings.GetTsFclTxTemplate(),
+		bindings.GetTsFclParamsTemplate(),
+		bindings.GetTsFclInterfaceTemplate(),
+	}
+
+	return &FclGenerator{
+		Templates: templates,
+	}
+}
+
+func NewFclJSGenerator() *FclGenerator {
+	templates := []string{
+		bindings.GetJsFclMainTemplate(),
+		bindings.GetJsFclScriptTemplate(),
+		bindings.GetJsFclTxTemplate(),
+		bindings.GetJsFclParamsTemplate(),
+	}
+
+	return &FclGenerator{
+		Templates: templates,
+	}
+}
+
+func (g *FclGenerator) Generate(flixString string, templateLocation string) (string, error) {
+	tmpl, err := parseTemplates(g.Templates)
+	if err != nil {
+		return "", err
+	}
+	if flixString == "" {
+		return "", fmt.Errorf("no flix template provided")
+	}
+	isLocal := !isUrl(templateLocation)
```

Comment:
id move this closer to where its used (line 54 maybe?) so it reads better contextually